### PR TITLE
Changes to initially load gameserver with location

### PR DESCRIPTION
**Pull request by hanzlamateen**
## Summary

Adds a Patch gameserver button in Instances section of admin panel. The button will preload selected location in free gameserver.

## Checklist
- [ ] Pre-push checks pass `npm run check`
  - [ ] Linter passing via `npm run lint`
  - [ ] Unit & Integration tests passing via `npm run test:packages`
  - [ ] Docker build process passing via `npm run build-client`
- [ ] If this PR is still a WIP, convert to a draft 
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

https://github.com/XRFoundation/XREngine/issues/5200

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers

_Reviewers for this PR_

### CODE REVIEW

File: packages/client-core/src/admin/components/Instance/PatchGameserver.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+  React.useEffect(() => {
+    if (location.created.value) {
+      closeViewModel(false)
+      setState({
+        ...state,
+        location: ''
+      })
+    }
+  }, [location.created.value])
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Instance/PatchGameserver.tsx:49]: with new changes which @speigg mentioned in discord the .value isn't required here anymore.

File: packages/gameserver/src/channels.ts [hunk 2]
```diff
@@ -612,6 +612,25 @@ export default (app: Application): void => {
...
+    const { id, ipAddress, port, locationId, sceneId } = params
+
+    if (app.instance && app.instance.id !== id) {
+      return
+    }
+
+    console.log('gameserver-load patched')
+    console.log({ id, ipAddress, port, locationId, sceneId })
+    const gsResult = await app.agonesSDK.getGameServer()
+    console.log(gsResult)
```
Comment (zulqarnainhanif) [inline packages/gameserver/src/channels.ts:628]: also please remove the logging which isn't required from the PR.

File: packages/server-core/src/networking/gameserver-provision/gameserver-provision-helper.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+export const patchGameserverLocation = async (app: Application, locationId) => {
+  if (app.k8DefaultClient) {
+    try {
+      const location = await app.service('location').find({
+        query: {
+          id: locationId
+        }
+      })
+
+      console.log(locationId)
```
Comment (zulqarnainhanif) [inline packages/server-core/src/networking/gameserver-provision/gameserver-provision-helper.ts:13]: please review the logging in this file as well.

**Review by HexaField — APPROVED** [review]

File: packages/client-core/src/admin/components/Instance/PatchGameserver.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                ))}
+              </Select>
+            </FormControl>
+          </Paper>
+          <DialogActions>
+            <Button className={classes.saveBtn} onClick={handleSubmit}>
+              Submit
+            </Button>
+            <Button onClick={handleClose(false)} className={classes.saveBtn}>
+              Cancel
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Instance/PatchGameserver.tsx:104]: {t('admin:components.setting.cancel')}

File: packages/client-core/src/admin/components/Instance/PatchGameserver.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                  <MenuItem value={el.id} key={i}>
+                    {el.name}
+                  </MenuItem>
+                ))}
+              </Select>
+            </FormControl>
+          </Paper>
+          <DialogActions>
+            <Button className={classes.saveBtn} onClick={handleSubmit}>
+              Submit
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Instance/PatchGameserver.tsx:101]: {t('admin:components.setting.save')} or create a new for submit button

File: packages/client-core/src/admin/components/Instance/PatchGameserver.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+      closeViewModel(false)
+    }
+  }
+
+  return (
+    <React.Fragment>
+      <Drawer anchor="right" classes={{ paper: classes.paperDrawer }} open={open} onClose={handleClose(false)}>
+        <Container maxWidth="sm" className={classes.marginTp}>
+          <DialogTitle id="form-dialog-title" className={classes.texAlign}>
+            Patch Gameserver
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Instance/PatchGameserver.tsx:72]: add `Patch Gameserver` in translation file as well

**Comment by zulqarnainhanif** [comment]
@hanzlamateen you can run `npm run format` and commit changes to get rid of pipeline failing

File: packages/server-core/src/networking/gameserver-provision/gameserver-provision-helper.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import { Application } from '../../../declarations'
+import { getFreeGameserver } from '../instance-provision/instance-provision.class'
+
+export const patchGameserverLocation = async (app: Application, locationId) => {
+  if (app.k8DefaultClient) {
```
Comment (barankyle) [inline packages/server-core/src/networking/gameserver-provision/gameserver-provision-helper.ts:5]: Do we need to check if k8DefaultClient exists? It seemed to work locally when I commented out this if statement; while there's not much use for it locally, I don't think it needs to explicitly not work there.

Comment (hanzlamateen) [reply packages/server-core/src/networking/gameserver-provision/gameserver-provision-helper.ts]: Agreed. Updated

**Comment by barankyle** [comment]
One last thing, I think the wording "Patch Gameserver" could be improved. Maybe "Preload gameserver"?

**Review by barankyle — APPROVED** [review]

**Pull request merged**